openedx/core/djangoapps/user_api/management/tests/test_email_opt_in_list.py

```diff
@@ -393,7 +412,11 @@ def _assert_output(self, output, *args, **kwargs):
         for user, course_id, opt_in_pref in args:
             self.assertIn({
                 "email": user.email.encode('utf-8'),
-                "full_name": user.profile.name.encode('utf-8'),
+                "full_name": (
+                    user.profile.name.encode('utf-8')
+                    if hasattr(user, 'profile')
```

Comment:
I mean, the method `_assert_output` is used in many tests.
We are getting `user.email` and `user.profile.fullname` in the method. If we pull the `fullname` logic out of this method then we have to add the `if` logic to all those methods.

One thing which I can understand if we pull it out, we would need that `if hasattr(user, 'profile')` above the method so that It does break on fullname when we pass it in the args. 

What do you think?